File: lab2/lab3.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {


    long long n1;
    cout << "Please enter a positive integer greater than 1:  ";
    cin >> n1;
    cout << endl;

    long long steps = 0;
    cout << n1;

    while (n1 > 1) {
        if (n1 % 2 == 0) {
            n1 = n1 / 2;
        } else {
            n1 = n1 * 3 + 1;
        }
        steps++;
        cout << " -> " << n1;
    }
    cout << "\nTotal steps: " << steps << "\n\n";


    int n2;
    cout << "Please enter a number between 10 and 100: ";
    cin >> n2;
    cout << endl;

    while (n2 < 10 || n2 > 100) {
        cout << "Invalid input. Please enter a number between 10 and 100:  ";
        cin >> n2;
        cout << endl;
    }

    int fizz_count = 0;
    int buzz_count = 0;
    int fizzbuzz_count = 0;

    for (int i = 1; i <= n2; i++) {
        if (i % 7 == 0) {
            cout << "(" << i << " is skipped)\n";
            continue;
        }

        if (i % 3 == 0 && i % 5 == 0) {
            cout << "FizzBuzz\n";
            fizzbuzz_count++;
        } else if (i % 3 == 0) {
            cout << "Fizz\n";
            fizz_count++;
        } else if (i % 5 == 0) {
            cout << "Buzz\n";
            buzz_count++;
        } else {
            cout << i << "\n";
        }
    }

    cout << "--- Summary ---\n";
    cout << "Fizz count : " << fizz_count << "\n";
    cout << "Buzz count : " << buzz_count << "\n";
    cout << "FizzBuzz count: " << fizzbuzz_count << "\n\n";


    int pattern_size;
    cout << "Please enter a number between 3 and 9:  ";
    cin >> pattern_size;
    cout << endl;
    for (int i = 1; i <= 2 * pattern_size - 1; i++) {
        int k = pattern_size - abs(pattern_size - i);

        for (int j = 0; j < k; j++) {
            cout << "*";
        }
        cout << endl;
    }

    return 0;
}

File: lab2/lab3.py
def main():

    n1 = int(input("Please enter a positive integer greater than 1: "))
    steps = 0
    print(n1, end="")

    while n1 > 1:
        if n1 % 2 == 0:
            n1 = n1 // 2
        else:
            n1 = n1 * 3 + 1
        steps += 1
        print(f" → {n1}", end="")
    print(f"\nTotal steps: {steps}\n")


    n2 = int(input("Please enter a number between 10 and 100: "))
    while not (10 <= n2 <= 100):
        n2 = int(input("Invalid input. Please enter a number between 10 and 100: "))

    fizz_count = 0
    buzz_count = 0
    fizzbuzz_count = 0

    for i in range(1, n2 + 1):
        if i % 7 == 0:
            print(f"({i} is skipped)")
            continue

        if i % 3 == 0 and i % 5 == 0:
            print("FizzBuzz")
            fizzbuzz_count += 1
        elif i % 3 == 0:
            print("Fizz")
            fizz_count += 1
        elif i % 5 == 0:
            print("Buzz")
            buzz_count += 1
        else:
            print(i)

    print("--- Summary ---")
    print(f"Fizz count : {fizz_count}")
    print(f"Buzz count : {buzz_count}")
    print(f"FizzBuzz count: {fizzbuzz_count}\n")

    pattern_size = int(input("Please enter a number between 3 and 9: "))

    for i in range(1, 2 * pattern_size):
        k = pattern_size - abs(pattern_size - i)
        print("*" * k)


if __name__ == "__main__":
    main()
